compiler/qsc_qasm3/src/lib.rs

```diff
@@ -0,0 +1,595 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#![allow(dead_code)]
```

Comment:
Why are we allowing `dead_code` in the entire crate? Is this intentional or was it left behind after initial development?

Reply:
I had it turned on while initially developing. I also accidentally implemented extra features which aren't turned on and may be detected as dead code, but are valuable to keep as we progress the feature.

Reply:
I'll do a pass with it turned on and clean up anything that is really dead

Reply:
Note that you can add a reason to the allow attribute
```rust
#[allow(dead_code, reason="this will support future qiskit features")]
```